Add parallax support

perlindgren opened the pull request:
A minor update to support Tiled parallax information. 

The `asset/tiled_parallax.tmx` provides a map with three layers (Background, Middle, and Foreground), each with different parallax information (Middle being the default Tiled (1.0, 1.0)).

The `tests/test` contains an example:
```rust
fn test_parallax_layers() {
    let r = read_from_file_with_path(&Path::new("assets/tiled_parallax.tmx")).unwrap();
    for (i, layer) in r.layers.iter().enumerate() {
        match i {
            0 => {
                assert_eq!(layer.name, "Background");
                assert_eq!(layer.parallax_x, 0.5);
                assert_eq!(layer.parallax_y, 0.75);
            }
            1 => {
                assert_eq!(layer.name, "Middle");
                assert_eq!(layer.parallax_x, 1.0);
                assert_eq!(layer.parallax_y, 1.0);
            }
            2 => {
                assert_eq!(layer.name, "Foreground");
                assert_eq!(layer.parallax_x, 2.0);
                assert_eq!(layer.parallax_y, 2.0);
            }
            _ => panic!("unexpected layer"),
        }
    }
}
```

The version is bumped to 0.9.6 to indicate the extended information.

Thanks for a great crate by the way!

bjorn commented:
@perlindgren These changes look good, but unfortunately conflict with the recent reorganization done in c3867665183a7165856830f048c2a390f77e88b8. Could you rebase your additions?

Btw, this crate has recently moved to the "mapeditor" organization so that we can share maintenance between those that use this crate. Let me know if you're interested in helping with further development!

PieKing1215 commented:
Just a note, this doesn't work on object layers as is.
You basically just need to copy the changes into `objects.rs` as well for it to work.

perlindgren commented:
So I fixed it, since the changes were so few and small it was actually easier just to copy/paste my changes into the new repo structure (mainly its in the `layers.rs`.

@bjorn did you have anything in particular in mind regarding the repo development. I played around a bit with Rust Bevy, and implemented parallax scrolling there, but since the `rs-tiled` crate did not support parallax info extraction I just went for it.

alexdevteam commented on src/layers.rs:
```diff
@@ -67,14 +69,16 @@ impl Layer {
-            ],
+                ("parallaxx", parallax_x, |v:String| v.parse().ok()),
+                ("parallaxy", parallax_y, |v:String| v.parse().ok()),
+                ],
```
The indentation is incorrect here.

alexdevteam commented on src/layers.rs:
```diff
@@ -104,6 +108,8 @@ impl Layer {
             visible: v.unwrap_or(true),
             offset_x: ox.unwrap_or(0.0),
             offset_y: oy.unwrap_or(0.0),
+            parallax_x:px.unwrap_or(1.0),
```
There's a missing space here.

alexdevteam reviewed (commented):
Pretty good, there are just a few minor formatting issues.

bjorn commented:
> So I fixed it, since the changes were so few and small it was actually easier just to copy/paste my changes into the new repo structure (mainly its in the `layers.rs`).

That's great, thanks! What about also adjusting `objects.rs` as per @PieKing1215's comment? And of course these attributes also affect `ImageLayer`.

Or, rather than duplicating more stuff, we could merge this as-is for now (just addressing the minor code style adjustments) and instead look into refactoring the data structure into an enum? I think we can pretty much copy the structs from https://gitlab.com/flukejones/tiled-json-rs/-/blob/master/src/layer.rs#L79-112.

> @bjorn did you have anything in particular in mind regarding the repo development. I played around a bit with Rust Bevy, and implemented parallax scrolling there, but since the rs-tiled crate did not support parallax info extraction I just went for it.

We'll need to make sure this crate supports all features of TMX and preferably also supports deserializing JSON. Secondly I think we need to make it more convenient to use, while keeping an eye on performance. Is that what you were asking?

bjorn commented:
I've addressed the minor code style issues and merged `master` again. I think this can be squashed and merged now.

Regarding adding these to `ObjectGroup` and `ImageLayer`, I think we can do it in a separate change as mentioned earlier.

alexdevteam reviewed (commented):
Add changelog changes and it's good to go

bjorn commented:
I'm not sure why those newlines in `assets/tiled_parallax.tmx` changed again...

bjorn commented:
> Sorry for being so late, the notification didn't get through.

No problem, it was good to wait with this until you were done with #131. :-)
